onboard/cv/main.cpp

```diff
@@ -70,15 +71,65 @@ void write_curr_frame_to_disk(Mat rgb, Mat depth, int counter) {
 int main() {
   /*initialize camera*/
   Camera cam;
+  cam.grab();
   int j = 0;
-  double frame_time = 0;
   int counter_fail = 0;
   #if PERCEPTION_DEBUG
-    namedWindow("image", 1);
+    namedWindow("Obstacle");
     namedWindow("depth", 2);
   #endif
   disk_record_init();
 
+  //Video Stuff
+  TagDetector d1;
+  pair<Tag, Tag> tp;
+
+  //Create time value
+  time_t now = time(0);
+  char* ltm = ctime(&now);
+  string timeStamp(ltm);
+  Mat rgb;
+  Mat src = cam.image();
+  
+
+  #if AR_RECORD
+  //initializing ar tag videostream object
+  
+  Mat depth_img = cam.depth();
+  tp = d1.findARTags(src, depth_img, rgb);
+  Size fsize = rgb.size();
+  
+  string s = "artag_" + timeStamp + ".avi";
+
+  VideoWriter vidWrite(s, VideoWriter::fourcc('M','J','P','G'),10,fsize,true);
+
+  if(vidWrite.isOpened() == false)
+  {
+	  cout << "didn't open";
```

Comment:
**NIT:** would it not be more appropriate to write this to stderr and add a newline?